test/extensions/filters/http/cache/http_cache_implementation_test_common.cc

```diff
@@ -0,0 +1,749 @@
+    if (trailers != actual_trailers) {
+      return AssertionFailure() << "Expected trailers == " << trailers
+                                << "\n  Actual:  " << actual_trailers;
+    }
+  }
+  return AssertionSuccess();
+}
+
+// Simple flow of putting in an item, getting it, deleting it.
+TEST_P(HttpCacheImplementationTest, PutGet) {
+  const std::string request_path1("/name");
+  LookupContextPtr name_lookup_context = lookup(request_path1);
+  EXPECT_EQ(CacheEntryStatus::Unusable, lookup_result_.cache_entry_status_);
+
+  Http::TestResponseHeaderMapImpl response_headers{
+      {":status", "200"},
+      {"date", formatter_.fromTime(time_system_.systemTime())},
+      {"cache-control", "public,max-age=3600"}};
+
+  const std::string body1("Value");
+  ASSERT_THAT(insert(move(name_lookup_context), response_headers, body1), IsOk());
+  name_lookup_context = lookup(request_path1);
+  EXPECT_TRUE(expectLookupSuccessWithBodyAndTrailers(name_lookup_context.get(), body1));
+
+  const std::string& request_path2("/another-name");
+  LookupContextPtr another_name_lookup_context = lookup(request_path2);
+  EXPECT_EQ(CacheEntryStatus::Unusable, lookup_result_.cache_entry_status_);
+
+  const std::string new_body1("NewValue");
+  ASSERT_THAT(insert(move(name_lookup_context), response_headers, new_body1), IsOk());
+  EXPECT_TRUE(expectLookupSuccessWithBodyAndTrailers(lookup(request_path1).get(), new_body1));
+}
+
+TEST_P(HttpCacheImplementationTest, PrivateResponse) {
+  Http::TestResponseHeaderMapImpl response_headers{
+      {":status", "200"},
+      {"date", formatter_.fromTime(time_system_.systemTime())},
+      {"age", "2"},
+      {"cache-control", "private,max-age=3600"}};
+  const std::string request_path("/name");
+
+  LookupContextPtr name_lookup_context = lookup(request_path);
+  ASSERT_EQ(CacheEntryStatus::Unusable, lookup_result_.cache_entry_status_);
+
+  const std::string body("Value");
+  // We must make sure at cache insertion time, private responses must not be
+  // inserted. However, if the insertion did happen, it would be served at the
+  // time of lookup.
+  ASSERT_THAT(insert(std::move(name_lookup_context), response_headers, body), IsOk());
+
+  LookupContextPtr next_lookup = lookup(request_path);
+  ASSERT_TRUE(expectLookupSuccessWithBodyAndTrailers(next_lookup.get(), body));
+  next_lookup->onDestroy();
+}
+
+TEST_P(HttpCacheImplementationTest, Miss) {
+  LookupContextPtr name_lookup_context = lookup("/name");
+  ASSERT_EQ(CacheEntryStatus::Unusable, lookup_result_.cache_entry_status_);
+}
+
+TEST_P(HttpCacheImplementationTest, Fresh) {
+  const std::string time_value_1 = formatter_.fromTime(time_system_.systemTime());
+  const Http::TestResponseHeaderMapImpl response_headers = {
+      {"date", time_value_1}, {"cache-control", "public, max-age=3600"}};
+  // TODO(toddmgreer): Test with various date headers.
+  ASSERT_THAT(insert("/", response_headers, ""), IsOk());
+  time_system_.advanceTimeWait(Seconds(3600));
```

Comment:
This is pre-existing but it's unfortunate we have an unqualified function `Seconds` in this infrastructure used in some places, and absl::Seconds in others. It would be better IMO if the unqualified one had a name that wasn't a substring of absl::Seconds, and also conformed to Envoy naming conventions, e.g. `chronoSeconds(...)` or something.

Reply:
Looks like this isn't defined for the cache filter, it's a type alias in common/time.h.